media-picker/src/main/java/com/miguelgaeta/media_picker/MediaPicker.java

```diff
@@ -158,38 +160,49 @@ public static void startForDocuments(final Provider provider, final OnError onEr
     }
 
     /**
-     * @see #startForImageCrop(Provider, File, int, int, int, OnError)
+     * @see #startForImageCrop(Provider, Uri, int, int, OnError)
      */
-    public static void startForImageCrop(final Provider provider, final File file, int outputWidth, int outputHeight, int colorInt, final OnError onError) {
+    public static void startForImageCrop(final Provider provider, final File file, int maxOutputWidth, int maxOutputHeight, final OnError onError) {
+        startForImageCrop(provider, Uri.fromFile(file), maxOutputWidth, maxOutputHeight, onError);
+    }
 
-        startForImageCrop(provider, Uri.fromFile(file), outputWidth, outputHeight, colorInt, onError);
+    public static void startForImageCrop(final Provider provider, final File file, int maxOutputWidth, int maxOutputHeight, final OnError onError, final @Nullable UCrop.Options options) {
+        startForImageCrop(provider, Uri.fromFile(file), maxOutputWidth, maxOutputHeight, onError, options);
     }
 
     /**
      * Start activity for cropping.
      *
      * @param provider Source {@link Provider}.
      * @param uri Source file URI.
-     * @param outputWidth Cropped file output width.
-     * @param outputHeight Cropped file output height.
+     * @param maxOutputWidth Cropped file output width.
+     * @param maxOutputHeight Cropped file output height.
      * @param colorInt Cropping UI circle color.
      * @param onError Result callbacks.
      */
-    private static void startForImageCrop(final Provider provider, final Uri uri, int outputWidth, int outputHeight, int colorInt, final OnError onError) {
+    public static void startForImageCrop(final Provider provider, final Uri uri, int maxOutputWidth, int maxOutputHeight, final OnError onError) {
+      startForImageCrop(provider, uri, maxOutputWidth, maxOutputHeight, onError, null);
+    }
+
+    public static void startForImageCrop(final Provider provider, final Uri uri, int maxOutputWidth, int maxOutputHeight, final OnError onError, final @Nullable UCrop.Options options) {
         try {
-            final Uri captureFileURI = createTempImageFileAndPersistUri(provider);
+            final UCrop.Options cropOptions = options != null
+                ? options
+                : new UCrop.Options();
 
-            final CropImageIntentBuilder intentBuilder = new CropImageIntentBuilder(outputWidth, outputHeight, captureFileURI);
+            // Force JPEG format compression now while working out filename strategy.
+            cropOptions.setCompressionFormat(Bitmap.CompressFormat.JPEG);
 
-            intentBuilder.setSourceImage(uri);
-            intentBuilder.setDoFaceDetection(false);
-            intentBuilder.setOutlineCircleColor(colorInt);
-            intentBuilder.setOutlineColor(colorInt);
-            intentBuilder.setScaleUpIfNeeded(true);
+            final Context context = provider.getContext();
+            final Uri destUri = Uri.fromFile(new File(context.getFilesDir(), "cropped.jpeg"));
```

Comment:
This is where I need some help in this PR. When I use that, even though I am calling the `grantWriteAccessToURI` method, the cropping fails -- I believe during the step when our activity tries to read the result of what the cropping activity wrote to. This is a cheap sidestep of the problem to avoid spending too much time figuring out something that you probably already know the answer to.

tl;dr this is a stopgap til Monday when I can pick your brain a little :)